frame/nfts/src/types.rs

```diff
@@ -182,39 +198,101 @@ pub enum CollectionSetting {
 	UnlockedMetadata,
 	/// Attributes of this collection can be modified.
 	UnlockedAttributes,
+	/// The supply of this collection can be modified.
+	UnlockedMaxSupply,
 	/// When this isn't set then the deposit is required to hold the items of this collection.
 	DepositRequired,
 }
-pub(super) type CollectionSettings = BitFlags<CollectionSetting>;
 
-/// Wrapper type for `CollectionSettings` that implements `Codec`.
+/// Wrapper type for `BitFlags<CollectionSetting>` that implements `Codec`.
 #[derive(Clone, Copy, PartialEq, Eq, Default, RuntimeDebug)]
-pub struct CollectionConfig(pub CollectionSettings);
+pub struct CollectionSettings(pub BitFlags<CollectionSetting>);
 
-impl CollectionConfig {
-	pub fn all_settings_enabled() -> Self {
+impl CollectionSettings {
+	pub fn all_enabled() -> Self {
 		Self(BitFlags::EMPTY)
 	}
-	pub fn get_disabled_settings(&self) -> CollectionSettings {
+	pub fn get_disabled(&self) -> BitFlags<CollectionSetting> {
 		self.0
 	}
+	pub fn is_disabled(&self, setting: CollectionSetting) -> bool {
+		self.0.contains(setting)
+	}
+	pub fn from_disabled(settings: BitFlags<CollectionSetting>) -> Self {
+		Self(settings)
+	}
+}
+
+impl_codec_bitflags!(CollectionSettings, u64, CollectionSetting);
+
+#[derive(Clone, Copy, Encode, Decode, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub enum MintType<CollectionId> {
+	/// Only an `Issuer` could mint items.
+	Private,
```

Comment:
Is there a better term than `private`? Some people use the term private NFTs as being NFTs that you can own without anyone knowing about it. Maybe CreatorMintable or some such?

Reply:
Done, changed to `Issuer`